app/server/lib/MinIOExternalStorage.ts

```diff
@@ -5,20 +5,41 @@ import {IncomingMessage} from 'http';
 import * as fse from 'fs-extra';
 import * as minio from 'minio';
 
-// The minio typings appear to be quite stale. Extend them here to avoid
-// lots of casts to any.
-type MinIOClient = minio.Client & {
-  statObject(bucket: string, key: string, options: {versionId?: string}): Promise<MinIOBucketItemStat>;
-  getObject(bucket: string, key: string, options: {versionId?: string}): Promise<IncomingMessage>;
-  listObjects(bucket: string, key: string, recursive: boolean,
-              options: {IncludeVersion?: boolean}): minio.BucketStream<minio.BucketItem>;
-  removeObjects(bucket: string, versions: Array<{name: string, versionId: string}>): Promise<void>;
-};
-
-type MinIOBucketItemStat = minio.BucketItemStat & {
-  versionId?: string;
-  metaData?: Record<string, string>;
-};
+// The minio v8.0.0 typings are sometimes incorrect. Here are some workarounds.
+interface MinIOClient extends
+  // Some of them are not directly extendable, must be omitted first and then redefined.
+  Omit<Omit<Omit<minio.Client,
+  "listObjects">,
+  "getBucketVersioning">,
+  "removeObjects">
```

Comment:
Could this Omit chain be replaced by this simpler syntax? (have not checked though, tell me if I miss something)

```suggestion
  Omit<minio.Client, "listObjects" | "getBucketVersioning" | "removeObjects">
```

Reply:
Thanks for pointing this out!